specification/orbital/resource-manager/Microsoft.Orbital/aodg/preview/2024-03-01-preview/aodg.json

```diff
@@ -0,0 +1,1750 @@
+        }
+      ],
+      "description": "A site where the Orbital network is connected to a partner network."
+    },
+    "GlobalCommunicationsSiteListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/GlobalCommunicationsSite"
+          },
+          "description": "A list of global communications sites."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to get the next set of results.",
+          "readOnly": true
+        }
+      },
+      "description": "Response for an API service call that lists Global Communications Sites."
+    },
+    "EdgeSite": {
+      "type": "object",
+      "required": [
+        "properties"
+      ],
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "allOf": [
+            {
+              "$ref": "#/definitions/EdgeSitesProperties"
+            }
+          ],
+          "description": "The properties bag for this resource."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../../common-types/resource-management/v5/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "description": "A customer's reference to a global communications site site."
+    },
+    "EdgeSiteListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/EdgeSite"
+          },
+          "description": "A list of edge site resources."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to get the next set of results.",
+          "readOnly": true
+        }
+      },
+      "description": "Response for an API service call that lists Edge Sites."
+    },
+    "L2ConnectionsProperties": {
+      "type": "object",
+      "required": [
+        "edgeSite",
+        "edgeSitePartnerRouter",
+        "groundStation",
+        "groundStationPartnerRouter",
+        "vlanId"
+      ],
+      "properties": {
+        "provisioningState": {
+          "type": "string",
+          "readOnly": true,
+          "allOf": [
+            {
+              "$ref": "#/definitions/ProvisioningState"
+            }
+          ],
+          "description": "The current state of the resource's creation, deletion, or modification."
+        },
+        "circuitId": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Globally-unique identifier for this connection that is to be used as a circuit ID."
+        },
+        "edgeSite": {
+          "type": "object",
+          "x-ms-client-flatten": true,
+          "required": [
+            "id"
+          ],
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ],
+          "properties": {
+            "id": {
```

Comment:
I didn't understand the overall relationship between resourceTypes. For example, l2Connections is a tracked resource as defined here, and it has a property called "id" (which should be renamed to "edgeSiteResourceId") to an edge site resource. Then, you also have a Post Action on edge site resource to "list" all l2Connections for a given edge site resource. Why not model l2Connections as a child Resource to edge site resource?

Reply:
The resource relationship is that an L2 connection has a reference to a GroundStation resource and EdgeSite resource. It represents the source and sink respectively of the connection. Since it is technically a child of 2 tracked resources, we didn't model L2 connection as a child resource since it would need to have a single parent.

L2 connection has an EdgeSite object property and a GroundStation object property. Both include an ID property as a reference to the tracked resource. We keep the EdgeSite and GroundStation object property so that we may extend it in the future in case we need to include a property other than the ID.

An EdgeSite and GroundStation tracked resource, then has a reference to a globalCommunicationSite object. This globalCommunicationSite object is not a proxy or tracked resource, it is created internally.